packages/core-js/utils/ast/finders.js

```diff
@@ -1,63 +1,113 @@
 const { findAll } = require('solidity-ast/utils');
 
+/**
+ * Get the given contract by id on the given AST
+ * @param {number} contractId
+ * @param {import("solidity-ast").SourceUnit} astNode
+ * @returns {import("solidity-ast").ContractDefinition}
+ */
 function findContractNodeWithId(contractId, astNode) {
-  return Array.from(findAll('ContractDefinition', astNode)).find(
-    (contractDefiniton) => contractDefiniton.id === contractId
-  );
+  for (const contractDefiniton of findAll('ContractDefinition', astNode)) {
+    if (contractDefiniton.id === contractId) {
+      return contractDefiniton;
+    }
+  }
 }
 
+/**
+ * Get the given contract by name on the given AST
+ * @param {string} contractName
+ * @param {import("solidity-ast").SourceUnit} astNode
+ * @returns {import("solidity-ast").ContractDefinition}
+ */
 function findContractNodeWithName(contractName, astNode) {
-  return Array.from(findAll('ContractDefinition', astNode)).find(
-    (contractDefiniton) => contractDefiniton.name === contractName
-  );
-}
-
-function getContractNode(astNode) {
-  if (!astNode) {
-    return undefined;
+  for (const contractDefiniton of findAll('ContractDefinition', astNode)) {
+    if (contractDefiniton.name === contractName) {
+      return contractDefiniton;
+    }
   }
-
-  return Array.from(findAll('ContractDefinition', astNode))[0];
 }
```

Comment:
Removing this `getContractNode` function is the main part of this PR. Because before we were finding the first contract defined on the given contract file, and now we are always requiring the user to specify a contract name (before we were doing it also, but didn't use it always, so it could get bad results).